compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -264,7 +263,27 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
                             format!("{}", item_typo),
                         ))
                     } else {
-                        suggestion
+                        // try to give a suggestion for this pattern: `name = 1`, which is common in other languages,
+                        // if we find an `=` is followed, we suggest to use `let name = 1` for an assignment
+                        let sm = self.r.session.source_map();
+                        let next_span = sm.span_look_ahead(span, None, Some(100));
+                        let line_span = sm.span_extend_to_line(span);
+                        let ident_name = sm.span_to_snippet(span).unwrap();
+                        if sm.span_to_snippet(next_span).map_or(false, |s| s.eq("="))
+                            && sm.span_to_snippet(line_span).map_or(false, |s| {
+                                s.trim().starts_with(&ident_name)
+                                    && !s.contains("==")
+                                    && !s.contains("let")
+                            })
```

Comment:
I need to spend more time on the review, but I would like to change this. Ideally we would recover the current expression and look at its parent to confirm it is indeed an assignment, instead of doing adhoc span recreation and snippet parsing (because it is brittle and breaks in the face of macros). Having said that, even with the snippet approach there are a couple of functions that do these kind of things (IIRC, there's an `span_until_char` or similar).

Reply:
Yes, my original thought is finding out the parent expression and recovery from that, but I haven't found out the right position, then I found some parts in `diagnostics` are doing this parsing hack, I followed it :).

Reply:
emmm, we have it at here: https://github.com/rust-lang/rust/blob/6153d3cbe6abc74fb37e4ebe48cc825484fd6bbf/compiler/rustc_ast_passes/src/feature_gate.rs#L592

for expr: ` ident: type = rhs', we suggest to add `let`, can we expand this to `ident = rhs` ?